packit/utils/obs_helper.py

```diff
@@ -0,0 +1,287 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+from osc import conf, core
+
+from packit.config import PackageConfig
+from packit.config.aliases import DEPRECATED_TARGET_MAP
+
+logger = logging.getLogger(__name__)
+
+_API_URL = "https://api.opensuse.org"
+
+
+@dataclass(frozen=True)
+class XmlPathEntry:
+    """Representation of a path entry in the XML repository configuration of
+    OBS.
+
+
+    """
+
+    project: str
+    repository: str
+
+
+@dataclass(frozen=True)
+class Repository:
```

Comment:
```suggestion
class OBSRepository:
```

Just to be sure we don't get into any collision with another class (e.g. from `git`/`github`/...)